ui/components/component-library/button-base/button-base.scss

```diff
@@ -0,0 +1,78 @@
+.mm-button {
+  position: relative;
+  height: 2.5rem;
+  padding: 0;
+  border-radius: 999px;
+  cursor: pointer;
+  color: var(--color-text-default);
+  background-color: #dcdcdc;
+  vertical-align: middle;
+  user-select: none;
+
+  &:active,
+  &:hover {
+    color: var(--color-text-default);
+  }
+
+  &--content {
+    height: 100%;
+  }
+
+  &-size--auto {
+    height: auto;
+    background-color: transparent;
+    border-radius: 0;
+    vertical-align: top;
+    font-family: inherit;
+    font-weight: inherit;
+    font-size: inherit;
+    line-height: inherit;
+    letter-spacing: inherit;
+  }
+
+  &-size--sm {
+    height: 2rem;
+  }
+
+  &-size--md {
+    height: 2.5rem;
+  }
+
+  &-size--lg {
+    height: 3rem;
+  }
+
+  &--loading {
+    cursor: not-allowed;
+  }
+
+  &--loading .mm-button--content {
+    color: transparent;
+  }
+
+  &--disabled,
+  &:disabled {
+    opacity: 0.3;
+    cursor: not-allowed;
+  }
+}
+
+.mm-button.box--display-block {
+  width: 100%;
+}
```

Comment:
I'm not sure we want to use other component class names within components. Mainly because if that class name gets updated in that component(in this case it probably wouldn't but I don't think we want to break that pattern) it would break this component. It would be more orthogonal if we created a `&--block` className and added the styles there